openedx/openedx-platform
# [FC-0009] New actions menu for copy/pasting units in Studio (behind waffle flag)

## Description

This PR implements https://github.com/openedx/modular-learning/issues/19 .

Normally, the Studio outline view looks like this:

![Screenshot 2023-08-01 at 12 14 45 PM](https://github.com/openedx/edx-platform/assets/945577/8bd3d1cd-db46-429c-afc8-627950509d4d)

When the new Waffle flag `contentstore.enable_copy_paste_units` is enabled, most of the actions get pushed into the new "Actions menu":

![Screenshot 2023-08-01 at 12 16 47 PM](https://github.com/openedx/edx-platform/assets/945577/26fd6e81-571d-40a1-b996-469e5391fd84)

This is analogous to the change for components that happened in https://github.com/openedx/edx-platform/pull/31853

As you can see, when the new menu is enabled, there is also a new **Copy to Clipboard** action that can be used to copy/paste units, either within a course or among different courses.

## Supporting information

See ticket linked above.

## Testing instructions

1. Check out this branch, rebuild static assets, etc.
2. Go to a course outline page and make sure the action buttons look "normal"
3. Go to http://studio.local.overhang.io:8001/admin/waffle/flag/ or devstack equivalent, and add a new flag called `contentstore.enable_copy_paste_units` which is enabled for Everyone.
4. Go to a course outline page and observe the new actions menu. Try copying and pasting a unit, within a course and between different courses.
5. Make sure the "Paste Unit" button disappears when a regular (non-unit) component is copied to the clipboard and vice versa. Check that this happens instantly across all open tabs.

## Deadline

None

## Other information

Depends on https://github.com/openedx/edx-platform/pull/32812

Private-ref: MNG-3771
